### PR TITLE
Refactors and workarounds so testing of collect_stats.py can be enabled in CI

### PR DESCRIPTION
**Pull request by shunghsiyu**
This pull requests contains a series of commits to refactor/workaround several blockers that prevented us from testing collect_stat.py in CI.
- Remove access to sysfs CPU device in collect_stats.py 
- Disable ASAN leak detection when testing collect_stats.py
- Give the libasan in LD_PRELOAD when testing collect_stats.py

### CODE REVIEW

**Comment by dcermak** [comment]
For some reason the debug builds fail with clang & ASAN with:
```
==127==ASan runtime does not come first in initial library list; you should either link runtime to your application or manually preload it with LD_PRELOAD.
```
This could be related to the `LD_LIBRARY_PATH` hackery that I had to do in https://github.com/SUSE/phoebe/blob/0e679270fc66e99bf468098b49f3d08cc91bab88/tests/meson.build#L10 to get ASAN to work with clang.

**Comment by shunghsiyu** [comment]
Thanks for the pointer, it seems more hackery is needed to get that part working :disappointed: . I'm working on that right now.

I'll push the code soon, and let's see if there is some way to make testing `collect_stats.py` with address sanitizer less painful. Or whether we prefer simply skipping `collect_stats.py` test instead (when address sanitizer is used).

**Comment by shunghsiyu** [comment]
Hmm...there seems to be problems with pulling container images from registry.opensuse.org (e.g. `registry.opensuse.org/home/dancermak/containers/phoebe`), and that's why the tests are failing.

`podman pull registry.opensuse.org/home/dancermak/containers/phoebe:latest` works for me locally though, so it might be a Azure/GitHub action problem.

**Comment by shunghsiyu** [comment]
I'll add some comments and push again tomorrow to re-trigger GitHub action. (I don't see any button where I can ask GitHub to run the checks again)

**Comment by mvarlese** [comment]
> I'll add some comments and push again tomorrow to re-trigger GitHub action. (I don't see any button where I can ask GitHub to run the checks again)

Please don't. I kicked off the re-running of jobs

**Comment by mvarlese** [comment]
@shunghsiyu The job is still failing; this time though is not an infrastructure issue, the actual test seems to fail instead.

File: scripts/meson.build [hunk 1]
```diff
@@ -1,16 +1,20 @@
-prog_python = import('python').find_installation('python3', required : get_option('collector'), modules : ['cffi'])
+prog_python = import('python').find_installation(
+  'python3',
+  required : get_option('collector'),
+  modules : ['cffi'],
+  disabler : true,
+)
+collector_script = files(meson.current_source_dir() / 'collect_stats.py')
```
Comment (dcermak) [inline scripts/meson.build]:
```suggestion
collector_script = files('collect_stats.py')
```
Using `meson.current_source_dir()` in `files()` is not required.

Comment (shunghsiyu) [reply scripts/meson.build]: Thanks!

**Comment by shunghsiyu** [comment]
I forgot that UBSAN also required a runtime and that LD_LIBRARY_PATH need to be set when `CC=clang && b_sanitize=undefined`, that is now fixed.

**Comment by mvarlese** [comment]
Thanks for the patch!

**Pull request merged**